packages/fungible-token/bridge-fungible-token/src/main.sw

```diff
@@ -88,6 +92,20 @@ impl MessageReceiver for Contract {
                 let sub_id = message_data.token_id;
                 let asset_id = AssetId::from(sha256((contract_id(), sub_id)));
 
+                disable_panic_on_overflow();
+
+                let current_total_supply = storage.tokens_minted.get(asset_id).try_read().unwrap_or(0);
+                let new_total_supply = current_total_supply + amount;
+
+                if new_total_supply < current_total_supply {
+                    register_refund(message_data.from, message_data.token_address, message_data.token_id, message_data.amount);
+                    return;
+                }
+
+                enable_panic_on_overflow();
```

Comment:
Wondering what happens if we return above but do not reach this line. Do we continue to have disabled panics for overflow or are they re-enabled automatically in the subsequent calls?

Reply:
To the best of my knowledge, this function cannot be called from other contexts inside the contract (it acts as an external function in solidity). 

OTOH, I 'd assume that contract-to-contract calls cannot "poison" these kind of flag configurations.